atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/persistentlock/LockStoreTest.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    @Test
+    public void createsPersistedLocksTable() {
+        LockStore.create(kvs);
+        verify(kvs, atLeastOnce()).createTable(eq(AtlasDbConstants.PERSISTED_LOCKS_TABLE), any(byte[].class));
+    }
+
+    @Test
+    public void lockIsInitiallyOpen() {
+        Set<LockEntry> lockEntries = lockStore.allLockEntries();
+
+        assertThat(lockEntries, contains(LockStore.LOCK_OPEN));
+    }
+
+    @Test
+    public void noErrorIfLockOpenedWhileCreatingTable() {
+        doThrow(new CheckAndSetException("foo", null, null, ImmutableList.of())).when(kvs).checkAndSet(anyObject());
+
+        lockStore.populateTable(); // should not throw
+    }
+
+    @Test
+    public void canAcquireLock() throws Exception {
+        LockEntry lockEntry = lockStore.acquireLock(REASON);
+
+        assertThat(lockStore.allLockEntries(), contains(lockEntry));
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
+    public void canNotAcquireLockTwice() throws Exception {
+        lockStore.acquireLock(REASON);
+        lockStore.acquireLock(REASON);
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
+    public void canNotAcquireLockTwiceForDifferentReasons() throws Exception {
+        lockStore.acquireLock(REASON);
+        lockStore.acquireLock("other-reason");
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
+    public void canNotAcquireLockThatWasTakenOutByAnotherStore() throws Exception {
+        LockStore otherLockStore = LockStore.create(kvs);
+        otherLockStore.acquireLock("grabbed by other store");
+
+        lockStore.acquireLock(REASON);
+    }
+
+    @Test
+    public void releaseLockPopulatesStoreWithOpenValue() throws Exception {
+        LockEntry lockEntry = lockStore.acquireLock(REASON);
+        lockStore.releaseLock(lockEntry);
+
+        assertThat(lockStore.allLockEntries(), contains(LockStore.LOCK_OPEN));
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
+    public void canNotReleaseNonExistentLock() throws Exception {
+        LockEntry lockEntry = lockStore.acquireLock(REASON);
+
+        LockEntry otherLockEntry = ImmutableLockEntry.builder().from(lockEntry).lockId("42").reason("other").build();
+        lockStore.releaseLock(otherLockEntry);
+    }
+
+    @Test
+    public void canReleaseLockAndReacquire() throws Exception {
+        LockEntry lockEntry = lockStore.acquireLock(REASON);
+        lockStore.releaseLock(lockEntry);
+
+        lockStore.acquireLock(REASON);
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
+    public void canNotReacquireAfterReleasingDifferentLock() throws Exception {
+        LockEntry lockEntry = lockStore.acquireLock(REASON);
+
+        LockEntry otherLockEntry = ImmutableLockEntry.builder().from(lockEntry).lockId("42").reason("other").build();
+        try {
+            lockStore.releaseLock(otherLockEntry);
+        } catch (IllegalArgumentException e) {
```

Comment:
catch `PersistentLockIsTakenException`, Isnt that what the test `canNotReleaseNonExistentLock` implies?